### PR TITLE
Embed: fix paths that start with `/`

### PR DESCRIPTION
**Pull request by stsewd**
can't wait to get rid of docnames :D

Ref https://sentry.io/organizations/read-the-docs/issues/2360423193/?environment=production&project=161479

### CODE REVIEW

File: readthedocs/embed/views.py [hunk 1]
```diff
@@ -165,10 +165,7 @@ def get(self, request):
 
         # Update doc from path
         if path:
-            if path.endswith('/'):
-                doc = doc.path.rstrip('/')
-            else:
-                doc = path.split('.html', 1)[0]
+            doc = re.sub(r'(.+)\.html$', r'\1', path.strip('/'))
```
Comment (humitos) [inline readthedocs/embed/views.py:168]: I think we should use `regex` module here since the path comes from user's input.

Comment (humitos) [reply readthedocs/embed/views.py:168]: Also, this line could have a comment explaining what it does plus an example.

Comment (stsewd) [reply readthedocs/embed/views.py:168]: a regex is dangerous when the pattern comes from the user, not the string to be matched.

Comment (humitos) [reply readthedocs/embed/views.py:168]: ha! I always get confused with this 😅

**Review by humitos — APPROVED** [review]

File: readthedocs/embed/views.py [hunk 2]
```diff
@@ -163,7 +163,8 @@ def get(self, request):
                 status=status.HTTP_400_BAD_REQUEST
             )
 
-        # Update doc from path
+        # Generate the docname from path
+        # by removing the ``.html`` extension and trailing ``/``.
```
Comment (humitos) [inline readthedocs/embed/views.py:167]: 💯

**Pull request merged**